BIGIP: Add support for lsnpool (CGNAT)

PrashantSunkari opened the pull request:
@caphrim007 

#### What issues does this address?
Fixes #1317

#### What's this change do?
Adds support for following CGNAT related resources. These are needed to address basic cgnat usecase.
- [x] LSNpool collection and resource + tests
- [x] LSN log profile collection and resource + tests
- [x] Add missing ALG profile - IPsecALG

Helps with following user stories:
1. User is able to create LSN pool without lsn log profile
2. User is able to create LSN pool with lsn log profile
3. User is able to create a virtual and attach a lsn pool and ALG profile

#### Where should the reviewer start?
- This work adds new functionality and can be reviewed in any order.
- Recommended path: 
  1. LSN pool code(f5/bigip/tm/ltm/lsnpool.py) covered lsnpool + lsnpool log profile
  2. Looking at tests (f5/bigip/tm/ltm/test/functional/test_lsnpool.py)
  3. New profile added - f5/bigip/tm/ltm/profile.py

#### Any background context?
- I work at F5 networks in CGNAT test org.

jputrino committed:
- updated SUPPORT.md with standard F5 language
- updated README.md
- updated CONTRIBUTING.md

(cherry picked from commit ec6b3e047e1b4945765d3d62d7adef56b79925ea)

jputrino committed:
- updated SUPPORT.md with standard F5 language
- updated README.md
- updated CONTRIBUTING.md

(cherry picked from commit ec6b3e047e1b4945765d3d62d7adef56b79925ea)

jputrino committed:
- updated SUPPORT.md with standard F5 language
- updated README.md
- updated CONTRIBUTING.md

(cherry picked from commit ec6b3e047e1b4945765d3d62d7adef56b79925ea)

jputrino committed:
- updated SUPPORT.md with standard F5 language
- updated README.md
- updated CONTRIBUTING.md

(cherry picked from commit ec6b3e047e1b4945765d3d62d7adef56b79925ea)

swormke committed:
Updated copyrights in README and CONTRIBUTING files after cherry-pick from develop

swormke committed:
Updated copyrights in README and CONTRIBUTING files after cherry-pick from develop

swormke committed:
Updated copyrights in README and CONTRIBUTING files after cherry-pick from develop

swormke committed:
Updated copyrights in README and CONTRIBUTING files after cherry-pick from develop

swormke committed:
Merge pull request #151 from swormke/master-docs

Project files in the master branch

swormke committed:
Merge branch '0.1'

zancas committed:
Merge pull request #266 from swormke/master

Merge 0.1 into Master

zancas committed:
Merge pull request #284 from F5Networks/0.1

Bugfix Release 0.1.2

Commit:
Merge branch 'master' of https://github.com/F5Networks/f5-common-python into cgnat_support

Commit:
Merge branch 'development' of https://github.com/F5Networks/f5-common-python into cgnat_support

PrashantSunkari committed:
WIP: Adding support for CGNAT lsnpool

Commit:
WIP: Adding support for CGNAT lsnpool

Commit:
WIP: Adding support for CGNAT lsnpool

Commit:
WIP: Adding support for CGNAT lsnpool

Commit:
WIP: Adding support for CGNAT lsnpool

PrashantSunkari committed:
Merge pull request #1 from PrashantSunkari/correcting_user_info

WIP: Adding support for CGNAT lsnpool

PrashantSunkari committed:
Issues:
Fixes #1317

Problem: Add support for lsnpool(CGNAT)

Analysis:
Added two things:
- LSN log profile collection and resource + corresponding tests
- Add missing ALG profile - IPsecALG

Tests:
- Added functional tests for lsnpool and lsn log profile
- Test results against BIGIP with 14.0.0 build
cmd: py.test test_lsnpool.py --ignore=f5/ --bigip=172.27.77.41 -vs

test_lsnpool.py::TestLSNPool::test_create_no_args PASSED
test_lsnpool.py::TestLSNPool::test_create PASSED
test_lsnpool.py::TestLSNPool::test_create_with_logProfile_without_logPub PASSED
test_lsnpool.py::TestLSNPool::test_create_with_logProfile PASSED
test_lsnpool.py::TestLSNPool::test_refresh PASSED
test_lsnpool.py::TestLSNPool::test_update PASSED
test_lsnpool.py::TestLSNPool::test_exists PASSED
test_lsnpool.py::TestLSNPool::test_stats PASSED
test_lsnpool.py::TestLSNPool::test_create_with_nondefault_mode PASSED
test_lsnpool.py::TestLSNPool::test_create_with_invalid_mode PASSED
test_lsnpool.py::TestLSNPool::test_create_with_member PASSED
test_lsnpool.py::TestLSNPool::test_virtual_with_pool PASSED
test_lsnpool.py::TestLSNPool::test_collection PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_create_no_args PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_create PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_update PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_exists PASSED

PrashantSunkari committed:
Issues:
Fixes #1317

Problem: Add support for lsnpool(CGNAT)

Analysis:
Added two things:
- LSN log profile collection and resource + corresponding tests
- Add missing ALG profile - IPsecALG

Tests:
- Added functional tests for lsnpool and lsn log profile
- Test results against BIGIP with 14.0.0 build
cmd: py.test test_lsnpool.py --ignore=f5/ --bigip=172.27.77.41 -vs

test_lsnpool.py::TestLSNPool::test_create_no_args PASSED
test_lsnpool.py::TestLSNPool::test_create PASSED
test_lsnpool.py::TestLSNPool::test_create_with_logProfile_without_logPub PASSED
test_lsnpool.py::TestLSNPool::test_create_with_logProfile PASSED
test_lsnpool.py::TestLSNPool::test_refresh PASSED
test_lsnpool.py::TestLSNPool::test_update PASSED
test_lsnpool.py::TestLSNPool::test_exists PASSED
test_lsnpool.py::TestLSNPool::test_stats PASSED
test_lsnpool.py::TestLSNPool::test_create_with_nondefault_mode PASSED
test_lsnpool.py::TestLSNPool::test_create_with_invalid_mode PASSED
test_lsnpool.py::TestLSNPool::test_create_with_member PASSED
test_lsnpool.py::TestLSNPool::test_virtual_with_pool PASSED
test_lsnpool.py::TestLSNPool::test_collection PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_create_no_args PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_create PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_update PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_exists PASSED

PrashantSunkari committed:
Issues:
Fixes #1317

Problem: Add support for lsnpool(CGNAT)

Analysis:
Added two things:
- LSN log profile collection and resource + corresponding tests
- Add missing ALG profile - IPsecALG

Tests:
- Added functional tests for lsnpool and lsn log profile
- Test results against BIGIP with 14.0.0 build
cmd: py.test test_lsnpool.py --ignore=f5/ --bigip=172.27.77.41 -vs

test_lsnpool.py::TestLSNPool::test_create_no_args PASSED
test_lsnpool.py::TestLSNPool::test_create PASSED
test_lsnpool.py::TestLSNPool::test_create_with_logProfile_without_logPub PASSED
test_lsnpool.py::TestLSNPool::test_create_with_logProfile PASSED
test_lsnpool.py::TestLSNPool::test_refresh PASSED
test_lsnpool.py::TestLSNPool::test_update PASSED
test_lsnpool.py::TestLSNPool::test_exists PASSED
test_lsnpool.py::TestLSNPool::test_stats PASSED
test_lsnpool.py::TestLSNPool::test_create_with_nondefault_mode PASSED
test_lsnpool.py::TestLSNPool::test_create_with_invalid_mode PASSED
test_lsnpool.py::TestLSNPool::test_create_with_member PASSED
test_lsnpool.py::TestLSNPool::test_virtual_with_pool PASSED
test_lsnpool.py::TestLSNPool::test_collection PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_create_no_args PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_create PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_update PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_exists PASSED

PrashantSunkari committed:
Issues:
Fixes #1317

Problem: Add support for lsnpool(CGNAT)

Analysis:
Added two things:
- LSN log profile collection and resource + corresponding tests
- Add missing ALG profile - IPsecALG

Tests:
- Added functional tests for lsnpool and lsn log profile
- Test results against BIGIP with 14.0.0 build
cmd: py.test test_lsnpool.py --ignore=f5/ --bigip=172.27.77.41 -vs

test_lsnpool.py::TestLSNPool::test_create_no_args PASSED
test_lsnpool.py::TestLSNPool::test_create PASSED
test_lsnpool.py::TestLSNPool::test_create_with_logProfile_without_logPub PASSED
test_lsnpool.py::TestLSNPool::test_create_with_logProfile PASSED
test_lsnpool.py::TestLSNPool::test_refresh PASSED
test_lsnpool.py::TestLSNPool::test_update PASSED
test_lsnpool.py::TestLSNPool::test_exists PASSED
test_lsnpool.py::TestLSNPool::test_stats PASSED
test_lsnpool.py::TestLSNPool::test_create_with_nondefault_mode PASSED
test_lsnpool.py::TestLSNPool::test_create_with_invalid_mode PASSED
test_lsnpool.py::TestLSNPool::test_create_with_member PASSED
test_lsnpool.py::TestLSNPool::test_virtual_with_pool PASSED
test_lsnpool.py::TestLSNPool::test_collection PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_create_no_args PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_create PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_update PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_exists PASSED

PrashantSunkari committed:
Issues:
Fixes #1317

Problem: Fixing comments and white spaces

Analysis: N/A

Tests:

test_lsnpool.py::TestLSNPool::test_create_no_args PASSED
test_lsnpool.py::TestLSNPool::test_create PASSED
test_lsnpool.py::TestLSNPool::test_create_with_logProfile_without_logPub PASSED
test_lsnpool.py::TestLSNPool::test_create_with_logProfile PASSED
test_lsnpool.py::TestLSNPool::test_refresh PASSED
test_lsnpool.py::TestLSNPool::test_update PASSED
test_lsnpool.py::TestLSNPool::test_exists PASSED
test_lsnpool.py::TestLSNPool::test_stats PASSED
test_lsnpool.py::TestLSNPool::test_create_with_nondefault_mode PASSED
test_lsnpool.py::TestLSNPool::test_create_with_invalid_mode PASSED
test_lsnpool.py::TestLSNPool::test_create_with_member PASSED
test_lsnpool.py::TestLSNPool::test_virtual_with_pool PASSED
test_lsnpool.py::TestLSNPool::test_collection PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_create_no_args PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_create PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_update PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_exists PASSED

real    0m4.261s

PrashantSunkari committed:
Issues:
Fixes #1317

Problem: Fixing comments and white spaces

Analysis: N/A

Tests:

test_lsnpool.py::TestLSNPool::test_create_no_args PASSED
test_lsnpool.py::TestLSNPool::test_create PASSED
test_lsnpool.py::TestLSNPool::test_create_with_logProfile_without_logPub PASSED
test_lsnpool.py::TestLSNPool::test_create_with_logProfile PASSED
test_lsnpool.py::TestLSNPool::test_refresh PASSED
test_lsnpool.py::TestLSNPool::test_update PASSED
test_lsnpool.py::TestLSNPool::test_exists PASSED
test_lsnpool.py::TestLSNPool::test_stats PASSED
test_lsnpool.py::TestLSNPool::test_create_with_nondefault_mode PASSED
test_lsnpool.py::TestLSNPool::test_create_with_invalid_mode PASSED
test_lsnpool.py::TestLSNPool::test_create_with_member PASSED
test_lsnpool.py::TestLSNPool::test_virtual_with_pool PASSED
test_lsnpool.py::TestLSNPool::test_collection PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_create_no_args PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_create PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_update PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_exists PASSED

real    0m4.261s

PrashantSunkari committed:
Issues:
Fixes #1317

Problem: Fixing comments and white spaces

Analysis: N/A

Tests:

test_lsnpool.py::TestLSNPool::test_create_no_args PASSED
test_lsnpool.py::TestLSNPool::test_create PASSED
test_lsnpool.py::TestLSNPool::test_create_with_logProfile_without_logPub PASSED
test_lsnpool.py::TestLSNPool::test_create_with_logProfile PASSED
test_lsnpool.py::TestLSNPool::test_refresh PASSED
test_lsnpool.py::TestLSNPool::test_update PASSED
test_lsnpool.py::TestLSNPool::test_exists PASSED
test_lsnpool.py::TestLSNPool::test_stats PASSED
test_lsnpool.py::TestLSNPool::test_create_with_nondefault_mode PASSED
test_lsnpool.py::TestLSNPool::test_create_with_invalid_mode PASSED
test_lsnpool.py::TestLSNPool::test_create_with_member PASSED
test_lsnpool.py::TestLSNPool::test_virtual_with_pool PASSED
test_lsnpool.py::TestLSNPool::test_collection PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_create_no_args PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_create PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_update PASSED
test_lsnpool.py::TestLSNPoolLogProfile::test_exists PASSED

real    0m4.261s

PrashantSunkari committed:
removing unused import

PrashantSunkari committed:
removing unused import

PrashantSunkari committed:
removing unused import

PrashantSunkari commented:
For some reason - my PR seems have pulled in changes that are not related to #1317. May be I messed up something - please let me know if its recommended to start a new PR.

[root@opensource-client functional]# git branch -vv
* cgnat_support        094736a [origin/development: ahead 14] removing unused import
  development          b2f5e1b [origin/development] Merge pull request #1316 from caphrim007/bugfix.allow-asm-tasks-to-import-to-partition

caphrim007 commented:
@PrashantSunkari this PR needs to be rebased that why you have the other PRs in there.

caphrim007 committed:
Adds iRulesLX (ilx) provisioning resources (#1320)

Issues:
Fixes #1319

Problem:
This API did not exist and prevented iRulesLX from being provisioned
or deprovisioned

Analysis:
This patch adds the resource

Tests:

PrashantSunkari committed:
Merge branch 'cgnat_support' of https://github.com/PrashantSunkari/f5-common-python into cgnat_support

PrashantSunkari committed:
resolving origin/develop merge conflicts

PrashantSunkari commented:
Closing this request and started a new request -  https://github.com/F5Networks/f5-common-python/pull/1321
